telethon/network/mtprotosender.py

```diff
@@ -357,25 +358,24 @@ async def _reconnect(self, last_error):
         self._state.reset()
 
         retries = self._retries if self._auto_reconnect else 0
-        for attempt in retry_range(retries):
+        
+        attempt = 0
+        for attempt in retry_range(retries,reconnect=True):
```

Comment:
Missed space before `reconnect` argument.
Also don't use explicit naming of arguments unless it's really necessary.